HTAQ.md

```diff
@@ -0,0 +1,66 @@
+# HOW TO ASK QUESTIONS
+:heavy_exclamation_mark:**PLEASE READ FULLY**:heavy_exclamation_mark:
+## INTRO
+In the world of hackers, the kind of answers you get to your technical questions depends as much on the way you ask the questions as on the difficulty of developing the answer. This guide will teach you how to ask questions in a way more likely to get you a satisfactory answer.
+
+**It sometimes looks like we're reflexively rude to newbies and the ignorant. But this isn't really true. Please try to ask good questions :smile_cat:**
+
+## Before You Ask
+Before asking a question, do the following:
+1. Try to find an answer by searching the archives of the our [chats](https://github.com/babayK0/frontend/blob/master/HTAQ.md#where-to-ask).
+2. Try to find an answer by searching the Web.
+3. Try to find an answer by reading the manual.
+4. Try to find an answer by reading a [FAQ](https://github.com/kottans/frontend/blob/master/faq.md).
+5. Try to find an answer by inspection or experimentation.
+6. Try to find an answer by asking a skilled friend.
+7. You're a programmer! So try to find an answer by reading the source code.
+
+When you ask your question, display the fact that you have done these things first; this will help establish that you're not being a lazy sponge and wasting people's time. Better yet, display what you have learned from doing these things. We like answering questions for people who have demonstrated they can learn from the answers.
+
+Take your time. Do not expect to be able to solve a complicated problem with a few seconds of Googling. Sit back, relax and give the problem some thought before approaching experts. Trust us, they will be able to tell from your questions how much reading and thinking you did, and will be more willing to help if you come prepared. Don't instantly fire your whole arsenal of questions just because your first search turned up no answers (or too many).
+
+Prepare your question. Think it through. Hasty-sounding questions get hasty answers, or none at all. The more you do to demonstrate that having put thought and effort into solving your problem before seeking help, the more likely you are to actually get help.
+
+**Remember, [there is no such thing as a stupid question.](https://en.wikipedia.org/wiki/No_such_thing_as_a_stupid_question)**:wink:
+
+## When You Ask
+* **Write in clear, grammatical, correctly-spelled language**
+* **Use [Stack Overflow](https://stackoverflow.com/)** 
+
+    _There are many similar questions, so try [SEARCH](https://stackoverflow.com/search?q=) first and then Ask._
+
+* **Be precise and informative about your problem**
+
+     Describe the symptoms of your problem or bug carefully and clearly.
+     
+     Describe the environment in which it occurs (machine, OS, application, whatever).
```

Comment:
`(machine, OS, application, whatever)` - the browser will be enough. It is web development.